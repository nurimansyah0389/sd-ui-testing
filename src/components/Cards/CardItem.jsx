import React from 'react';
import PropTypes from 'prop-types';
import numeral from 'numeral';
import dayjs from 'dayjs';

import './style.css';

const handleChooseFlight = item => alert(`Flight ${item.flightName} selected!`)

const CardItem = ({ data }) => (
    <div className="Card__Item">
        <h2>{data.flightName}</h2>
        <div className="Item">
            <div className="Item__Info">
                <div className="Left">
                    <div className="Left__Logo"><img src="https://static.tiket.photos/image/upload/v1534836229/string/2018/08/21/93102e79-7442-458e-a227-789bd08eddb0c4f3f430c09d1128f1f0019aaaec2679.png" alt=""/></div>
                    <div className="Left__Departure">
                        <div>{dayjs(data.departureTime).format('hh:mm')}</div>
                        <div>{data.departureCode}</div>
                    </div>

                    <div className="Left__Duration">
                        <div>{Math.floor(data.tripDuration / 3600)}h 0m</div>
                        <div>Direct</div>
                    </div>

                    <div className="Left__Arrival">
                        <div>{dayjs(data.arrivalTime).format('hh:mm')}</div>
                        <div>{data.arrivalCode}</div>
                    </div>
                </div>
                <div className="Right">
                    <div className="Right__Price">
                        <span>IDR {numeral(data.price).format('0,0')}</span>
                        <span>/pax</span>
                    </div>
                </div>
            </div>
            <div className="Item__Action">
                <button className="Button Button_Primary" type="button" onClick={() => handleChooseFlight(data)}>Choose Flight</button>
            </div>
        </div>
    </div>
);
CardItem.propTypes = {
    data: PropTypes.shape({
        id: PropTypes.string,
        flightName: PropTypes.string,
        departureCode: PropTypes.string,
        departureTime: PropTypes.number,
        arrivalCode: PropTypes.string,
        arrivalTime: PropTypes.number,
        tripDuration: PropTypes.number,
        price: PropTypes.number
    })
};
export default CardItem;