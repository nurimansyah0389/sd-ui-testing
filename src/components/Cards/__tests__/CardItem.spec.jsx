import React from 'react';
import Renderer from 'react-test-renderer';

import CardItem from '../CardItem';

describe('Card Item Spec', () => {
    it('Should render without any error', () => {
        const item = {
            flightName: 'Garuda Indonesia',
            departureTime: Math.round(new Date().getTime() / 1000),
            arrivalTime: Math.round(new Date().getTime() / 1000) + 3600,
            departureCode: 'CGK',
            arrivalCode: 'SIN',
            tripDuration: 3600,
            price: 360000
        };
        const tree = Renderer.create(<CardItem data={item} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});