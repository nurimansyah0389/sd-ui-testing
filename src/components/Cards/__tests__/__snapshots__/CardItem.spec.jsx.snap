// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Item Spec Should render without any error 1`] = `
<div
  className="Card__Item"
>
  <h2>
    Garuda Indonesia
  </h2>
  <div
    className="Item"
  >
    <div
      className="Item__Info"
    >
      <div
        className="Left"
      >
        <div
          className="Left__Logo"
        >
          <img
            alt=""
            src="https://static.tiket.photos/image/upload/v1534836229/string/2018/08/21/93102e79-7442-458e-a227-789bd08eddb0c4f3f430c09d1128f1f0019aaaec2679.png"
          />
        </div>
        <div
          className="Left__Departure"
        >
          <div>
            11:36
          </div>
          <div>
            CGK
          </div>
        </div>
        <div
          className="Left__Duration"
        >
          <div>
            1
            h 0m
          </div>
          <div>
            Direct
          </div>
        </div>
        <div
          className="Left__Arrival"
        >
          <div>
            11:36
          </div>
          <div>
            SIN
          </div>
        </div>
      </div>
      <div
        className="Right"
      >
        <div
          className="Right__Price"
        >
          <span>
            IDR 
            360,000
          </span>
          <span>
            /pax
          </span>
        </div>
      </div>
    </div>
    <div
      className="Item__Action"
    >
      <button
        className="Button Button_Primary"
        onClick={[Function]}
        type="button"
      >
        Choose Flight
      </button>
    </div>
  </div>
</div>
`;
